components/api/src/test/java/com/hotels/styx/api/messages/FullHttpRequestTest.java

```diff
@@ -0,0 +1,442 @@
+/**
+ * Copyright (C) 2013-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.api.messages;
+
+import com.google.common.collect.ImmutableMap;
+import com.hotels.styx.api.HttpRequest;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import rx.observers.TestSubscriber;
+
+import java.net.InetSocketAddress;
+
+import static com.hotels.styx.api.HttpCookie.cookie;
+import static com.hotels.styx.api.HttpHeader.header;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_LENGTH;
+import static com.hotels.styx.api.HttpHeaderNames.COOKIE;
+import static com.hotels.styx.api.HttpHeaderNames.HOST;
+import static com.hotels.styx.api.HttpRequest.Builder.put;
+import static com.hotels.styx.api.Url.Builder.url;
+import static com.hotels.styx.api.messages.FullHttpRequest.get;
+import static com.hotels.styx.api.messages.FullHttpRequest.patch;
+import static com.hotels.styx.api.messages.FullHttpRequest.post;
+import static com.hotels.styx.api.messages.FullHttpRequest.toStreamingHttpRequest;
+import static com.hotels.styx.support.matchers.IsOptional.isAbsent;
+import static com.hotels.styx.support.matchers.IsOptional.isValue;
+import static com.hotels.styx.support.matchers.MapMatcher.isMap;
+import static io.netty.buffer.Unpooled.copiedBuffer;
+import static io.netty.handler.codec.http.HttpMethod.DELETE;
+import static io.netty.handler.codec.http.HttpMethod.GET;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_0;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static java.lang.String.valueOf;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static rx.Observable.just;
+
+
+public class FullHttpRequestTest {
+    @Test
+    public void encodesToStreamingHttpRequest() throws Exception {
```

Comment:
Here we need to have tests to demonstrate that all HTTP request attributes (URL, path, protocol, headers) are converted accordingly by the call to `toStreamingHttpRequest()`. Not just the request body.